src/gpuarray/buffer.h

```diff
@@ -634,8 +637,16 @@ GPUARRAY_PUBLIC gpucontext *gpukernel_context(gpukernel *k);
  */
 #define GA_CTX_PROP_MAXLSIZE2 17
 
+/**
+ * Get the vector of collective ops for the context.
+ *
+ * Type: `const gpuarray_comm_ops *`
+ */
+#define GA_CTX_PROP_COMM_OPS  18
```

Comment:
Do you have some time to chat? Because I find myself mimicking the way that you've wrapped things for blas in order to be consistent with your design style, but if you say that you' ve changed your mind, I think we should discuss this. Can you tell me what happened: what did you like and what you did not? Things like that.